tests/agentic_system/tasks/test_dataloader.py

```diff
@@ -0,0 +1,262 @@
+import pytest
+import pandas as pd
+import numpy as np
+
+from dspy_litl_agentic_system.tasks.prism_lookup import PrismKey, PrismLookup
+
+
+@pytest.fixture
+def sample_data():
+    """Create sample PRISM data for testing."""
+    return pd.DataFrame({
+        'drug': ['DrugA', 'DrugB', 'DrugC', ' DrugD ', 'DrugA'],
+        'cell_line': ['CellX', 'CellY', 'CellZ', 'CellX', 'CellY'],
+        'ic50': [1.5, 2.3, 0.8, 3.1, 1.9],
+        'auc': [0.75, 0.82, 0.65, 0.91, 0.78],
+        'viability': [0.25, 0.18, 0.35, 0.09, 0.22]
+    })
```

Comment:
Consider keeping all fixtures within a file `conftest.py` which is often the convention when using `pytest`. `pytest` will automatically load these with the same references you have despite them being in another file. The benefit is that you always know where your fixtures are and multiple test modules can use the same fixture pool.